### DISCUSSION TITLE
Database Triggers FutureCall?

### DISCUSSION THREAD

**Discussion by tricaricom** [serverpod/serverpod#2663] [Q&A] [answered]
Is there any way to run a FutureCall based on a database action?  For example, when a new user record is created via Apple/Google sign-in, I would like to execute a FutureCall so I can do some additional work on the server without having to call an endpoint.

Thanks!

**Comment by Isakdl** [comment, marked as answer]
We should probably expand the docs with some examples for this, but here you have all available configurations and callbacks available in the auth module: https://docs.serverpod.dev/concepts/authentication/setup#configure-authentication

```dart
auth.AuthConfig.set(auth.AuthConfig(  
    onUserCreated: (Session session, UserInfo userInfo) {
       //Register your future call here. or any other action you want to take
    },
));
```

But with that said, we do not have any direct hook into the database that will trigger for any arbitrary table/column action today.

**Comment by tricaricom** [comment]
Thanks!
